kratos/utilities/intersection_utilities.h

```diff
@@ -233,6 +237,118 @@ class IntersectionUtilities
         return true;
     }
 
+    /**
+     * @brief Calculates the line to line intersection (shortest line). If line is length 0, it is considered a point and therefore there is intersection (3D version)
+     * @details Calculate the line segment PaPb that is the shortest route between two lines P1P2 and P3P4. Calculate also the values of mua and mub where
+     *    Pa = P1 + mua (P2 - P1)
+     *    Pb = P3 + mub (P4 - P3)
+     *    http://paulbourke.net/geometry/pointlineplane/
+     * @param rSegment1 The first segment
+     * @param rSegment2 The second segment
+     * @tparam TGeometryType The geometry type
+     * @return Return empty points array if no solution exists. Otherwise returns the line intersection points array
+     */
+    template<class TGeometryType>
+    static PointerVector<Point> ComputeShortestLineBetweenTwoLines(
+        const TGeometryType& rSegment1,
+        const TGeometryType& rSegment2
+        )  
+    {
+        // Zero tolerance
+        const double zero_tolerance = std::numeric_limits<double>::epsilon();
+
+        // Check geometry type
+        KRATOS_ERROR_IF(rSegment1.GetGeometryType() != GeometryData::KratosGeometryType::Kratos_Line3D2) << "The geometry type is not correct, it is suppossed to be a linear 3D line" << std::endl;
```

Comment:
1. If the implementation only supports `Line3D2` the template argument must be in my opinion the point type. 

2. On top of this, maybe I'm missing something, but isn't the `Line2D2` a particular in-plane case of the 3D one? As far as I understand the same algorithm should work in both cases.

Reply:
Nope, in 2D the closest line is always a perpendicular line or intersection, I tried this is simpler than specialize (I am the first that if I could use constexpr or templates I would use it)

Reply:
Yes, we already discussed about specializing the implementation. My comment was regarding the error thrown, why are we forbidding using this with 2D lines? Shouldn't it work as in the 3D case with intersected or aligned lines?

Reply:
Technically, yes, I can add them. I will see if I can check geometry family instead

Reply:
Okay, @rubenzorrilla last change should work both 2D and 3D

Reply:
Perfect. This was exactly my point.